_im-plugin/similarity.md

```diff
@@ -0,0 +1,622 @@
+
+For example, if you try to change from `DFR` similarity to `boolean` similarity, you'll get an error like:
+
+```
+"Unknown settings for similarity of type [boolean]: [normalization.h2.c, normalization, after_effect, basic_model]"
+```
+
+To resolve this, explicitly set the old parameters to `null` when updating:
+
+```json
+PUT /product_catalog/_settings
+{
+  "index": {
+    "similarity": {
+      "default": {
+        "type": "boolean",
+        "basic_model": null,
+        "after_effect": null,
+        "normalization": null,
+        "normalization.h2.c": null
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuring a built-in similarity on the field level
+
+OpenSearch provides built-in similarity algorithms (`BM25` and `boolean`) that can be used directly in field mappings. To configure similarity on the field level, apply the similarity to the field when creating the index. The following example applies a `boolean` similarity to the `content` field:
+
+```json
+PUT /blog_posts_2/
+{
+  "mappings": {
+    "properties": {
+      "content": {
+        "type": "text",
+        "similarity": "boolean"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+In this example, only the `content` field uses boolean similarity. Other fields in the index will use the default BM25 similarity.
+
+## Configuring a custom similarity
+
+For more advanced use cases, you can define named custom similarities that can be applied to specific fields. This approach provides greater flexibility when different fields in your index need different scoring algorithms.
+
+Configuring named custom similarities requires a two-step process:
+
+1. Define the similarity in index settings with a custom name
+2. Apply the similarity to specific fields in mappings
+
+### Step 1: Define a custom similarity
+
+The following example creates an index with a custom DFR similarity configuration:
+
+```json
+PUT /blog_posts
+{
+  "settings": {
+    "index": {
+      "similarity": {
+        "blog_similarity": {
+          "type": "DFR",
+          "basic_model": "g",
+          "after_effect": "l",
+          "normalization": "h2",
+          "normalization.h2.c": "3.0"
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Step 2: Apply the custom similarity to fields
+
+After defining the similarity, you must explicitly apply it to specific fields in your mappings. Unlike default similarity, custom similarities are not automatically applied to all fields:
+
+```json
+PUT /blog_posts/_mapping
+{
+  "properties": {
+    "content": {
+      "type": "text",
+      "similarity": "blog_similarity"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+In this example, only the `content` field uses the custom `blog_similarity`. Other fields in the index (if any) continue using the default BM25 similarity unless explicitly configured otherwise.
```

Comment:
```suggestion
In this example, only the `content` field uses the custom `blog_similarity`. Other fields in the index (if any) continue using the default `BM25` similarity unless explicitly configured otherwise.
```